src/Microsoft.Azure.CosmosEventSourcing/Stores/DefaultEventStore.Def.cs

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) IEvangelist. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Azure.CosmosEventSourcing.Items;
+using Microsoft.Azure.CosmosRepository;
+
+namespace Microsoft.Azure.CosmosEventSourcing.Stores;
+
+internal partial class DefaultEventStore<TEventItem> :
+    IEventStore<TEventItem> where TEventItem : DefaultEventItem
+{
+    private readonly IRepository<TEventItem> _repository;
+
+    public DefaultEventStore(IRepository<TEventItem> repository) =>
+        _repository = repository;
```

Comment:
It might make sense to get the `IReadOnlyRepository` and `IWriteOnlyRepository` from DI, and then use each in the respective _*.Read.cs_ and _*.Write.cs_ files.

Reply:
Yeah I agree I like that idea! I will raise this as another PR along with the work on EventItem